frontend/src/components/LineChartComponents/LineChart.tsx

```diff
@@ -0,0 +1,760 @@
+import {RoundedRectangle} from '@amcharts/amcharts5/.internal/core/render/RoundedRectangle';
+import {Color, color} from '@amcharts/amcharts5/.internal/core/util/Color';
+import {DataProcessor} from '@amcharts/amcharts5/.internal/core/util/DataProcessor';
+import {IXYChartSettings, XYChart} from '@amcharts/amcharts5/.internal/charts/xy/XYChart';
+import {IDateAxisSettings} from '@amcharts/amcharts5/.internal/charts/xy/axes/DateAxis';
+import {AxisRendererX} from '@amcharts/amcharts5/.internal/charts/xy/axes/AxisRendererX';
+import {AxisRendererY} from '@amcharts/amcharts5/.internal/charts/xy/axes/AxisRendererY';
+import {XYCursor} from '@amcharts/amcharts5/.internal/charts/xy/XYCursor';
+import am5locales_en_US from '@amcharts/amcharts5/locales/en_US';
+import am5locales_de_DE from '@amcharts/amcharts5/locales/de_DE';
+import {darken, useTheme} from '@mui/material/styles';
+import Box from '@mui/material/Box';
+import {useTranslation} from 'react-i18next';
+import {Dictionary} from '../../util/util';
+import React from 'react';
+import {Localization} from 'types/localization';
+import useRoot from 'components/shared/Root';
+import {useConst} from 'util/hooks';
+import useXYChart from 'components/shared/LineChart/Chart';
+import useDateAxis from 'components/shared/LineChart/DateAxis';
+import useValueAxis from 'components/shared/LineChart/ValueAxis';
+import {useDateSelectorFilter} from 'components/shared/LineChart/Filter';
+import useDateAxisRange from 'components/shared/LineChart/AxisRange';
+import {useLineSeriesList} from 'components/shared/LineChart/LineSeries';
+import {GroupFilter} from 'types/group';
+import {ScenarioList} from 'types/scenario';
+import {LineSeries} from '@amcharts/amcharts5/.internal/charts/xy/series/LineSeries';
+import {LineChartData} from 'types/lineChart';
+
+interface LineChartProps {
+  /** Optional unique identifier for the chart. Defaults to 'chartdiv'. */
+  chartId?: string;
+
+  /** Array of line chart data points to be plotted on the chart. */
+  lineChartData: LineChartData[] | undefined;
+
+  /** The currently selected date in the chart in ISO format (YYYY-MM-DD). */
+  selectedDate: string;
+
+  /** Callback function to update the selected date in the chart. */
+  setSelectedDate: (date: string) => void;
+
+  /**
+   * Optional callback function to get the x-coordinate position of the reference date in the chart.
+   * This can be used for positioning elements in relation to the reference date.
+   */
+  setReferenceDayBottom?: (docPos: number) => void;
+
+  /** Optional minimum date for the chart in ISO format (YYYY-MM-DD). */
+  minDate?: string | null;
+
+  /** Optional maximum date for the chart in ISO format (YYYY-MM-DD). */
+  maxDate?: string | null;
+
+  /** Optional currently selected scenario identifier. */
+  selectedScenario?: number | null;
+
+  /** Optional array of active scenario identifiers. These scenarios will be displayed on the chart. */
+  activeScenarios?: number[] | null;
```

Comment:
I think we can get rid of these. Ideally all the info about the drawn lines should already be in the `lineChartData` and we shouldn't need the logic in here.